crates/core/src/client/mod.rs

```diff
@@ -443,11 +437,12 @@ impl<T: JsonRpcTransport + Send + Sync> KakarotEthApi<T> for KakarotClient<JsonR
     ) -> Result<TokenBalances, EthApiError<T::Error>> {
         let entrypoint: Felt252Wrapper = keccak256("balanceOf(address)").try_into()?;
         let entrypoint: FieldElement = entrypoint.into();
-        let felt_address = FieldElement::from_str(&address.to_string()).map_err(|e| {
-            EthApiError::<T::Error>::OtherError(anyhow::anyhow!("Failed to convert address to FieldElement: {}", e))
-        })?;
+
+        let add: Felt252Wrapper = address.into();
```

Comment:
is there something preventing from doing 

```
let address: Felt252Wrapper = address.into()
```
?

Reply:
`let address = Felt252Wrapper::from(address);`

Reply:
We are actually using the `address` couple of lines lower, which needs to be a `H160`